frontend/src/metabase/public/containers/PublicDashboard/WithPublicDashboardEndpoints.tsx

```diff
@@ -0,0 +1,33 @@
+import type { ComponentType } from "react";
+import type { WithRouterProps } from "react-router";
+
+import {
+  setEmbedDashboardEndpoints,
+  setPublicDashboardEndpoints,
+} from "metabase/services";
+import type { DashboardId } from "metabase-types/api";
+
+export const WithPublicDashboardEndpoints = <T extends WithRouterProps>(
```

Comment:
Maybe we could even put these in the `routes` for `embed` and `public` respectively on the `onEnter` hook. but that's for another PR I think